rasa/nlu/selectors/embedding_response_selector.py

```diff
@@ -0,0 +1,198 @@
+        "mu_neg": -0.4,  # should be -1.0 < ... < 1.0 for 'cosine'
+        # flag: if true, only minimize the maximum similarity for
+        # incorrect intent labels
+        "use_max_sim_neg": True,
+        # scale loss inverse proportionally to confidence of correct prediction
+        "scale_loss": True,
+        # regularization parameters
+        # the scale of L2 regularization
+        "C2": 0.002,
+        # the scale of how critical the algorithm should be of minimizing the
+        # maximum similarity between embeddings of different intent labels
+        "C_emb": 0.8,
+        # dropout rate for rnn
+        "droprate": 0.2,
+        # visualization of accuracy
+        # how often to calculate training accuracy
+        "evaluate_every_num_epochs": 20,  # small values may hurt performance
+        # how many examples to use for calculation of training accuracy
+        "evaluate_on_num_examples": 0,  # large values may hurt performance,
+        # selector config
+        # name of the intent for which this response selector is to be trained
+        "retrieval_intent": None,
+    }
+    # end default properties (DOC MARKER - don't remove)
+
+    def _load_selector_params(self, config: Dict[Text, Any]):
+        self.retrieval_intent = config["retrieval_intent"]
+
+    def _load_params(self) -> None:
+        super(ResponseSelector, self)._load_params()
+        self._load_selector_params(self.component_config)
+
+    def process(self, message: "Message", **kwargs: Any) -> None:
+        """Return the most likely response and its similarity to the input."""
+
+        label, label_ranking = self.predict_label(message)
+
+        selector_key = (
+            self.retrieval_intent
+            if self.retrieval_intent
+            else DEFAULT_OPEN_UTTERANCE_TYPE
+        )
+
+        logger.debug(
+            "Adding following selector key to message property: {}".format(selector_key)
+        )
+
+        prediction_dict = {"response": label, "ranking": label_ranking}
+
+        self._set_message_property(message, prediction_dict, selector_key)
+
+    @staticmethod
+    def _set_message_property(
+        message: "Message", prediction_dict: Dict[Text, Any], selector_key: Text
+    ):
+        if message.get(MESSAGE_SELECTOR_PROPERTY_NAME):
+            message_selector_properties = message.get(MESSAGE_SELECTOR_PROPERTY_NAME)
+            message_selector_properties[selector_key] = prediction_dict
+            message.set(
+                MESSAGE_SELECTOR_PROPERTY_NAME,
+                message_selector_properties,
+                add_to_output=True,
+            )
+        else:
+            message_selector_properties = {selector_key: prediction_dict}
+            message.set(
+                MESSAGE_SELECTOR_PROPERTY_NAME,
+                message_selector_properties,
+                add_to_output=True,
+            )
+
+    def preprocess_data(self, training_data):
```

Comment:
let's rename it, I think it clashes with `process` method name

Reply:
changed it to `preprocess_train_data`